build/charts/antrea/crds/bgppolicy.yaml

```diff
@@ -0,0 +1,129 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: bgppolicies.crd.antrea.io
+spec:
+  group: crd.antrea.io
+  versions:
+    - name: v1alpha1
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          required:
+            - spec
+          properties:
+            spec:
+              type: object
+              required:
+                - nodeSelector
+                - advertisements
```

Comment:
I noticed that trying to create the following policy with `kubectl apply` failed:
```yaml
apiVersion: crd.antrea.io/v1alpha1
kind: BGPPolicy
metadata:
  name: test
spec:
  nodeSelector: {}
```
with: `The BGPPolicy "test" is invalid: spec.advertisements: Required value`

This is expected because we are defining `advertisements` as a required property (if creating the policy from a Golang client, it would not be an issue because there is no difference between an empty struct and an omitted struct).

This means that to create a policy that does not advertise anything, we need:
```yaml
apiVersion: crd.antrea.io/v1alpha1
kind: BGPPolicy
metadata:
  name: test
spec:
  nodeSelector: {}
  advertisements: {}
```

Personally I think it is ok, because:
1) it does not affect the Golang client
2) it forces a user to express that they do not want to advertise anything explicitly, and ensures that they did not simply forgot to provide the field

But I wanted to make sure that this was the intended behavior and that everyone approves of it.

Reply:
How about removing the requirement since it is a legal case for running BGP process only,  not advertising any routes, not establishing peer relationships with remote routers.

Reply:
I know it's a legal case, but it was already possible to express it with `advertisements: {}`. Now we have 2 ways of expressing the same thing (`advertisements: {}` and  `advertisements: null`) and I am not sure this is an improvement. IMO, the previous version may have been better.

Reply:
cc @tnqn on `*Advertisenents` vs `Advertisements`
personally I have a preference for the non-pointer version

Reply:
+1 to non-pointer, given we don't need to distinguish unset from zero value, making it pointer would add the burden for nil check.

Reply:
Thanks for explaining this. I will revert back to non-pointer.